### DISCUSSION TITLE
[5.x] Boolean values stored as strings?

### DISCUSSION THREAD

**Discussion by rob-c-baker** [craftcms/cms#16222] [Q&A] [answered]
I ran into this while querying data in one of our Craft 5 installs. This commit: https://github.com/craftcms/cms/commit/8e34669640eb7e4e4fd89dd95e7cc111bffc0eac changed the way booleans are stored in the JSON `content` blob in the `elements_sites` DB table so they are stored as the strings `'true'` & `'false'`. A comment suggests that it's the only reliable way to compare values.

Does anyone know why this is the case?

There must be more to this - something to do with type juggling maybe but on what level? i.e. is it to work around a database issue or something?

_Can't find nay documentation relating to it so thought it might be helpful to have it here so it might pop up when searching._

**Comment by brandonkelly** [comment, marked as answer]
It’s been a while, but I believe this was necessary for referencing boolean values in SQL conditions.

The data is still stored as boolean `true`/`false` values, though.
